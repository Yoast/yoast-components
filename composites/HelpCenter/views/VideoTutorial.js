import React from "react";
import styled from "styled-components";
import YouTubeVideo from "../../basic/YouTubeVideo";
import PropTypes from "prop-types";

const Section = styled.section`
	display: flex;
	padding: 1em;
`;

const VideoTutorialPlaceholder = styled( Section )`
	flex-wrap: wrap;
`;

const VideoTextPanel = styled( Section )`
	min-width: 220px;
	max-width: 550px;
	flex: 50% 0;
`;

const StyledYouTubeVideo = styled( YouTubeVideo )`
	flex: 50% 0;
`;

/**
 * Creates a VideoTutorial component that displays a YouTube video.
 *
 * @param {Object} props The props to use for the component.
 * @returns {ReactElement} The rendered VideoTutorial component.
 *
 * @constructor
 */
export default function VideoTutorial( props ) {
	return (
		<VideoTutorialPlaceholder>
			<StyledYouTubeVideo	src={ props.src } title={ props.title } />
			<VideoTextPanel>
				{ props.description }
			</VideoTextPanel>
		</VideoTutorialPlaceholder>
	);
}

VideoTutorial.propTypes = {
	src: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
	description: PropTypes.string.isRequired,
};
