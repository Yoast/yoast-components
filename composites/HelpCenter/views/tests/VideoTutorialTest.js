import React from "react";
import renderer from "react-test-renderer";

import VideoTutorial from "../VideoTutorial.js";

test( "the VideoTutorial component matches the snapshot", () => {
	const component = renderer.create(
		<VideoTutorial src="https:/www.youtube.com/embed/bIgcj_pPIbw" title="Video title" description="Video description"/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

