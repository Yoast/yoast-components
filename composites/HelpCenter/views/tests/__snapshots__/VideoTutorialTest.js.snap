// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the VideoTutorial component matches the snapshot 1`] = `
<section
  className="sc-htpNat hqgsEe sc-bwzfXH gwKLSd"
>
  <iframe
    allowFullScreen={true}
    className="sc-bdVaJa bDWFJH"
    frameBorder={0}
    height={315}
    src="https:/www.youtube.com/embed/bIgcj_pPIbw"
    title="Video title"
    width={560}
  />
  <section
    className="sc-bxivhb foTyhW sc-bwzfXH gwKLSd"
  >
    Video description
  </section>
</section>
`;
